docs/reference/contributing/target/Watchdog.md

```diff
@@ -1,25 +1,45 @@
-<h3 id="watchdog-port">Watchdog</h3>
+### Watchdog
 
-[Include a brief description here.]
+The Watchdog HAL API provides a low-level interface to the Independent Watchdog Timer of a target. Implementing the Watchdog API is not mandatory, but by implementing it, you can use the hardware watchdog timer to detect and recover from computer malfunctions.
+
+Implement the Reset Reason API when implementing the Watchdog API. The Reset Reason API allows you to detect at boot time whether a watchdog caused the last reset.
 
 #### Assumptions
 
 ##### Defined behavior
 
-[Include any defined behavior in bullet format here.]
+- Sleep and debug modes don't stop the watchdog from counting down.
+- The function `hal_watchdog_init` is safe to call repeatedly. It must not do anything if you have already initialized the watchdog timer.
```

Comment:
Query: Is "watchdog timer" on this line an API or something else?

Reply:
It refers to the hardware watchdog timer, but it relates to the API function `hal_watchdog_init`.

Maybe "The function `hal_watchdog_init` is safe to call repeatedly. The function's implementation must not do anything if the hardware watchdog timer has already been initialised by `hal_watchdog_init`."